server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -329,6 +329,28 @@ public boolean offer(E e) {
             }
         }
 
+        // The following three methods, put, add, and timed offer, are
+        // overridden to workaround a JDK bug introduced in JDK 21.0.2
+        // https://bugs.openjdk.org/browse/JDK-8323659
+        @Override
+        public void put(E e) {
+            boolean added = super.offer(e);
+            assert added;
```

Comment:
This works as LTQ is unbounded, offer will always succeed.

Reply:
:+1: maybe add this as a code comment